hexa/pipelines/schema/mutations.py

```diff
@@ -328,6 +328,10 @@ def resolve_upload_pipeline(_, info, **kwargs):
         return {"success": False, "errors": ["INVALID_TIMEOUT_VALUE"]}
     except PermissionDenied:
         return {"success": False, "errors": ["PERMISSION_DENIED"]}
+    except IntegrityError as e:
+        if "unique_pipeline_version_name" in str(e):
```

Comment:
Not fan of hardcoding the constraint name here, do we want to extract the name in a config file and avoid duplicate definition ?

Reply:
I agree. You could replace this by checking the presence of a version with the same name and returning the specific error code if it `.exists()`. I'd still catch the IntegrityError to return a broader error type. 🤷🏻

Reply:
Checking with `.exists()` is not concurrent-safe : if 2 requests check for existence at the same time they will both succeed the check and then one of them will fail with `IntegrityError`, so we have inconsistency in error management (same error can be caught in the `exists()` check and sometimes in IntegrityError). That's why I have not followed that route. 

If we think that we are OK with that in OH, in can implement it that way, if not we need to rely on the DB or any other transactional mechanism

Reply:
As of now we do not put lot of focus on this since we do not have the throughput that would make it plausible to happen. The result would not result in an incoherent DB neither but more of a broader error message to the user. It's a trade-off we chose until now. 

I'm not a fan of having hardcoded strings neither. I can propose you multiple solutions:
- We use your approach with a check in the string
- We check if the cause of the exception is `psycopg2.errors.UniqueViolation` (`e.__cause__` I think)
- We use a constant on the model to pass the name of the constraint

Do the one you prefer.

Reply:
I followed the constant approach with also a check of the instance class of `__cause__`

(Interestingly [at least for my learning], I tried to override `save()` to call for `full_clean()` which is checking out-of-the-box the unique constraint before inserting in the DB. But it also checks for other attributes that should be non-null but for now are passed null from the SDK . Also this setup is  very similar to checking the IntegrityError so that's why I decided to keep it)